syntax = "proto2";

package rv.protos;

import "rastervision/protos/machine_learning.proto";
import "rastervision/protos/scene.proto";

message PredictConfig {
    message ObjectDetectionOptions {
        /*
            If predicted boxes have an IOA (intersection over area) greater than
            merge_thresh, then they are merged into a single box during
            postprocessing. This is needed since the sliding window approach
            results in some false duplicates.
        */
        optional float merge_thresh = 1 [default=0.5];

        // Predicted boxes are only output if their score is above score_thresh.
        optional float score_thresh = 2 [default=0.5];
    }

    message ClassificationOptions {

    }

    message Options {
        // The URI of the model file output by the train command.
        optional string model_uri = 1;

        // The prediction chip_size needs to be the same as the chip_size
        // used when training the model.
        optional int32 chip_size = 3;

        optional bool debug = 4 [default=true];
        // Root of dir to write debug files to.
        optional string debug_uri = 7;

        oneof ml_options_type {
            ObjectDetectionOptions object_detection_options = 5;
            ClassificationOptions classification_options = 6;
        }
    }

    required MachineLearning machine_learning = 1;
    repeated Scene scenes = 4;
    required Options options = 5;
}
